hawaii.py

```diff
@@ -14,4 +17,228 @@ def main():
+
+        if complex_area not in dict1:
+            dict1[complex_area] = {}
+            dict1[complex_area]["Total Classrooms"] = total_classes
+            dict1[complex_area]["Ventilated Classrooms"] = ventilated_classes
+            dict1[complex_area]["Classrooms Lacking Ventilation"] = ventilation_gap
+                  
+        dict1[complex_area]["Complex Area"] = excel_data["Complex Area"].iloc[i]
+        dict1[complex_area]["Pull Date"] = excel_data["Pull Date"].iloc[i]
+        dict1[complex_area]["Ventilated Classrooms"] += int(excel_data["M3 Ventilated Classrooms"].iloc[i])
+        dict1[complex_area]["Classrooms Lacking Ventilation"] += int(excel_data["M3 Ventilation Gap"].iloc[i])
+        dict1[complex_area]["Total Classrooms"] += int(excel_data["M3 Total Classrooms"].iloc[i])
+
+    # append to pandas dataframe
+    for row in dict1:
+        df = df.append(dict1[row], ignore_index=True)
+    
+    df.to_csv("out/hawaii_ventilation_" + datetime.now().strftime('%Y%m%d') + ".csv", index=False)
+
+
+# Can Accomodate 20-21 Enrollment (full time schedule)?
+def social_distancing():
+    column_names = []
+    excel_data = pd.read_excel("test.xlsx", sheet_name="Metric 2")
+
+    df = pd.DataFrame(columns=column_names)
+
+    for i in range(0, len(excel_data["Complex Area"]) - 1):
+        row = {"Complex Area": excel_data["Complex Area"].iloc[i], 
+                "Complex": excel_data["Complex"].iloc[i], 
+                "School Code": excel_data["School Code"].iloc[i], 
+                "Name": excel_data["Name"].iloc[i], 
+                "Pull Date": excel_data["Pull Date"].iloc[i], 
+                "Can Accomodate Social Distancing?": excel_data["Can Accomodate 20-21 Enrollment (full time schedule)?"].iloc[i]}
+        df = df.append(row, ignore_index=True)
+
+    df.to_csv("out/hawaii_social_distancing_" + datetime.now().strftime('%Y%m%d') + ".csv", index=False)
+
+
+def device_gap():
+    column_names = []
+    excel_data = pd.read_excel("test.xlsx", sheet_name="Metric 11")
+
+    dict1 = {}
+    df = pd.DataFrame(columns=column_names)
+
+    # loop through and combine complex areas
+    for i in range(0, len(excel_data["Complex Area"]) - 1):
+        complex_area = excel_data["Complex Area"].iloc[i]
+
+        if complex_area not in dict1:
+            dict1[complex_area] = {}
+            dict1[complex_area]["Number Devices"] = 0
+            dict1[complex_area]["Device Gap"] = 0
+            dict1[complex_area]["Number of Devices For Learning"] = 0
+            dict1[complex_area]["Device Gap / Lacking"] = 0
+
+        dict1[complex_area]["Complex Area"] = excel_data["Complex Area"].iloc[i]
+        dict1[complex_area]["Pull Date"] = excel_data["Pull Date"].iloc[i]
+        dict1[complex_area]["Number of Devices For Learning"] += int(excel_data["Metric 11 Enrl"].iloc[i])
+        dict1[complex_area]["Device Gap / Lacking"] += int(excel_data["Metric 11 Device Gap"].iloc[i])
```

Comment:
It might be better to retain the same column schema and granularity level as the input Excel sheet. Instead of aggregating by complex area in this script, it's simpler to store the disaggregated data. Users can aggregate the data further downstream if needed.

I'm not sure if there's a specific format that the Hawaii data should be stored as, so disregard this comment if the data already meets the specified format.